src/main/java/resources/AuthResource.java

```diff
@@ -45,4 +51,34 @@ public Response validateToken(@QueryParam("token") String token) {
 			return Response.status(Response.Status.GONE).build();
 		}
 	}
-}
+
+	@GET
+	@UnitOfWork
+	@Path("activate/{activationLink}")
+	public Person checkIfPersonWithActivationLinkExists (@PathParam("activationLink")String activationLink) {
+		java.util.Optional<Person> person = personDao.getUserByActivationLink(activationLink);
+
+		return person.orElseThrow(() -> new WebApplicationException(Response.Status.NO_CONTENT));
+	}
+
+	@PUT
+	@UnitOfWork
+	@Path("/activate")
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Response activatePerson (Person person) {
+		java.util.Optional<Person> personToBeActivated = personDao.getById(person.getId());
+
+		if(personToBeActivated.isPresent()) {
+			personToBeActivated.get().setActivationCode(null);
+			personToBeActivated.get().setActive(true);
+			try {
+				personToBeActivated.get().setPassword(passwordStore.createHash(person.getPassword()));
+			} catch (TdpRecruitmentPasswordStore.CannotPerformOperationException e) {
+				e.printStackTrace();
```

Comment:
Return an error which indicates a problem from the catch block -- otherwise a successful response will be returned even though an exception has been thrown!